src/drive/web/modules/filelist/FileIcon.jsx

```diff
@@ -1,18 +1,19 @@
 import React from 'react'
+import flag from 'cozy-flags'
 import FileIconMime from 'drive/web/modules/filelist/FileIconMime'
 import FileIconShortcut from 'drive/web/modules/filelist/FileIconShortcut'
-import { ImageLoader } from 'components/Image'
+import FileImageLoader from 'cozy-ui/transpiled/react/FileImageLoader'
 import styles from 'drive/styles/filelist.styl'
 
 const FileIcon = ({ file, size, isEncrypted }) => {
   const isImage = file.class === 'image'
   const isShortcut = file.class === 'shortcut'
-
-  if (isImage)
+  const shoudUseThumbnailsForPDF = flag('drive.thumbnails-pdf.enabled')
+  if (isImage || (shoudUseThumbnailsForPDF && file.class === 'pdf'))
```

Comment:
I think this feature needs a little time to be tested. If you don't have it, tell me :-)

Reply:
What do you want to test here?

Reply:
```
import React from 'react'
import { render } from '@testing-library/react'
import FileIcon from './FileIcon'

jest.mock('cozy-flags', () => () => true)
jest.mock('cozy-ui/transpiled/react/FileImageLoader', () => () => (
  <div data-testid="FileImageLoader" />
))

describe('FileIcon', () => {
  it('should return file image loader when file is image', () => {
    // Given
    const file = { class: 'image' }

    // When
    const { getByTestId } = render(<FileIcon file={file} />)

    // Then
    expect(getByTestId('FileImageLoader')).toBeInTheDocument()
  })

  it('should return file image loader when file is pdf', () => {
    // Given
    const file = { class: 'pdf' }

    // When
    const { getByTestId } = render(<FileIcon file={file} />)

    // Then
    expect(getByTestId('FileImageLoader')).toBeInTheDocument()
  })
})
```

Reply:
or https://github.com/cozy/cozy-drive/commit/9d449f4a9a78c177eb4a3a0d9f4336f3866bf86e

Reply:
Typically what I call "testing for testing sake" but hey you're in charge here now so I cherry-picked your commit.